main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -1352,31 +1352,72 @@ object Resolver {
             b => ResolvedAst.Expr.ArrayLength(b, loc)
           }
 
-        case NamedAst.Expr.StructNew(name, fields, region, loc) =>
-          val fieldsVal = traverse(fields) {
-            case (f, e) =>
-              val eVal = visitExp(e, env0)
-              mapN(eVal) {
-                case e => (f, e)
+        case NamedAst.Expr.StructNew(sym, fields, region, loc) =>
+          lookupStruct(sym, env0, ns0, root) match {
+            case Result.Ok(st) =>
+              val providedFieldNames = fields.map {case (k, _) => k}.toSet
+              val expectedFieldNames = st.fields.map(_.sym).toSet
+              val extraFields = providedFieldNames.diff(expectedFieldNames).map(_.name)
```

Comment:
Instead of one error for all the fields, could we do multiple errors, one for each extra field?